spec.md

```diff
@@ -2387,6 +2389,58 @@ module DUT :
   connect y, p
 ```
 
+## Property Assignments
+
+Connections between property typed expressions (see [@sec:property-types]) are
+not supported in the `connect`{.firrtl} statement (see [@sec:connects]).
+
+Instead, property typed expressions are assigned with the `propassign`{.firrtl}
+statement.
+
+Property typed expressions have the normal rules for flow (see [@sec:flows]),
+but otherwise use a stricter, simpler algorithm than `connect`{.firrtl}. In
+order for a property assignment to be legal, the following conditions must hold:
+
+1. The left-hand and right-hand side expressions must be references (see
+   [@sec:references]).
+
+2. The left-hand and right-hand side expressions must be of property types.
+
+3. The types of the left-hand and right-hand side expressions must be the same.
+
+4. The flow of the left-hand side expression must be sink.
+
+5. The flow of the right-hand side expression must be source.
+
+6. The left-hand side expression must be used as the left-hand side in at most
+   one property assignment.
+
+7. The property assignment may not occur within a conditional scope.
+
+Note that property types are not legal for any expressions with duplex flow.
+
+There is no notion of a connection algorithm (see
+[@sec:the-connection-algorithm]) for sub-elements of property typed expressions.
+Unlike vectors and bundles, property types are always passive (see
+[@sec:passive-types]).
```

Comment:
The connection algorithm discusses how you connect aggregates (albeit trivially, but exhaustively specified to avoid ambiguity).  I'm not sure what the second sentence means as aggregate types are orthogonal to passivity.

Reply:
Let me re-word this. I really just want to say property types are passive and must be connected as a whole.

Reply:
I changed this to "Unlike vectors and bundles, property types must be connected as a whole." I'm not sure if I even want to say this here, since we aren't even talking about property aggregates yet. Maybe I can just remove this section?

Reply:
i think its confusing since if we can't assign to a literal (you haven't created that yet) and you haven't defined aggregates, its unclear what you CAN assign to/from other than a pass-through port, so it does seem a bit early for this statement

Reply:
Yeah, let me just cut this. I was thinking it would be helpful to constrast to how connects work, but really i think we should just precisely define how propassigns work (which is quite limited for now, as you point out).

Reply:
I think I'm going to cut "There is not notion of last connect semantics" as well--it's redundant with the list above, and in the spirit of just defining propassign without comparing it to connect, i think this makes it more simple without losing information.